### PR TITLE
fix: Remove cancel changes button from a disruption create from a template

### PR DESCRIPTION
**Pull request by DeannaSharma**

### CODE REVIEW

File: site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx [hunk 1]
```diff
@@ -68,9 +69,9 @@ const TypeOfConsequence = (props: ConsequenceTypePageProps): ReactElement => {
                                     role="button"
                                     href={
                                         returnToTemplateOverview
-                                            ? `${queryParams["return"] as string}/${pageState.disruptionId || ""}${
-                                                  isTemplate ? "?template=true" : ""
-                                              }`
+                                            ? `${queryParams["return"] as string}/${
```
Comment (ljones92) [inline site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx:72]:
```suggestion
                                            ? `${returnPath}/${
```

File: site/pages/api/delete-disruption.api.ts [hunk 2]
```diff
@@ -1,19 +1,33 @@
...
+    ERROR_PATH,
+    VIEW_ALL_TEMPLATES_PAGE_PATH,
+} from "../../constants";
 import { deletePublishedDisruption, getDisruptionById } from "../../data/dynamo";
-import { isDisruptionFromTemplate, redirectTo, redirectToError } from "../../utils/apiUtils";
+import { redirectTo, redirectToError } from "../../utils/apiUtils";
 import { canPublish, getSession } from "../../utils/apiUtils/auth";
 import logger from "../../utils/logger";
 
+const getRedirectPath = (template: string, returnPath: string) => {
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/api/delete-disruption.api.ts:14]: done

File: site/pages/api/delete-disruption.api.ts [hunk 3]
```diff
@@ -35,7 +49,7 @@ const deleteDisruption = async (req: NextApiRequest, res: NextApiResponse): Prom
 
         await deletePublishedDisruption(disruption, id, session.orgId, template === "true");
 
-        redirectTo(res, template || isFromTemplate ? VIEW_ALL_TEMPLATES_PAGE_PATH : DASHBOARD_PAGE_PATH);
+        redirectTo(res, getRedirectPath(template as string, returnPath as string));
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/api/delete-disruption.api.ts:52]: done

File: site/pages/disruption-detail/[disruptionId].page.tsx [hunk 4]
```diff
@@ -345,7 +346,7 @@ const DisruptionDetail = ({
             {popUpState && csrfToken ? (
                 <DeleteConfirmationPopup
                     entityName={`the ${popUpState.name}`}
-                    deleteUrl={`/api/delete-${popUpState.name}${disruption.template ? "?template=true" : ""}`}
+                    deleteUrl={`/api/delete-disruption${disruption.template ? "?template=true" : ""}`}
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/disruption-detail/[disruptionId].page.tsx:349]: I'll revert this change back as PR 284 supersedes this.

With the old code, when we delete a template, the deleteUrl takes you to /delete-template which throws a 404.

File: site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx [hunk 5]
```diff
@@ -19,7 +19,7 @@ import { getStateUpdater, returnTemplateOverview, showCancelButton } from "../..
 const title = "Create Consequences";
 const description = "Create Consequences page for the Create Transport Disruptions Service";
 
-export interface ConsequenceTypePageProps extends PageState<Partial<ConsequenceType>> {}
+export type ConsequenceTypePageProps = PageState<Partial<ConsequenceType>>;
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx:22]: I think my IDE auto-corrected this - gave me the below error. 
Screenshot 2023-09-27 at 08 03 36

File: site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx [hunk 6]
```diff
@@ -46,6 +46,7 @@ const CreateConsequenceNetwork = (props: CreateConsequenceNetworkProps): ReactEl
     const returnToTemplateOverview = returnTemplateOverview(queryParams);
 
     const isTemplate = (queryParams["template"] as string) || "";
+    const returnPath = (queryParams["return"] as string) || "";
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx:49]: done

File: site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx [hunk 7]
```diff
@@ -173,9 +174,9 @@ const CreateConsequenceNetwork = (props: CreateConsequenceNetworkProps): ReactEl
                                 role="button"
                                 href={
                                     returnToTemplateOverview
-                                        ? `${queryParams["return"] as string}/${pageState.disruptionId || ""}${
-                                              isTemplate ? "?template=true" : ""
-                                          }`
+                                        ? `${queryParams["return"] as string}/${
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx:177]: done

File: site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx [hunk 8]
```diff
@@ -173,9 +174,9 @@ const CreateConsequenceNetwork = (props: CreateConsequenceNetworkProps): ReactEl
...
+                                              pageState.disruptionId || ""
+                                          }?template=true`
                                         : `${queryParams["return"] as string}/${pageState.disruptionId || ""}`
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/create-consequence-network/[disruptionId]/[consequenceIndex].page.tsx:180]: done

File: site/pages/create-disruption/[disruptionId].page.tsx [hunk 9]
```diff
@@ -56,7 +61,9 @@ const CreateDisruption = (props: DisruptionPageProps): ReactElement => {
     const [addValidityClicked, setAddValidityClicked] = useState(false);
 
     const queryParams = useRouter().query;
-    const displayCancelButton = showCancelButton(queryParams);
+    const displayCancelChangesButton = showCancelButton(queryParams);
+    const returnPath = (queryParams["return"] as string) || "";
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/create-disruption/[disruptionId].page.tsx:65]: done

File: site/pages/create-disruption/[disruptionId].page.tsx [hunk 10]
```diff
@@ -248,15 +255,15 @@ const CreateDisruption = (props: DisruptionPageProps): ReactElement => {
     return (
         <BaseLayout title={title} description={description} errors={props.errors}>
             <CsrfForm
-                action={`/api/create-disruption${queryParams["template"] ? "?template=true" : ""}`}
+                action={`/api/create-disruption${isTemplate == "true" ? "?template=true" : ""}`}
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/create-disruption/[disruptionId].page.tsx:258]: done

File: site/pages/create-disruption/[disruptionId].page.tsx [hunk 11]
```diff
@@ -637,27 +644,36 @@ const CreateDisruption = (props: DisruptionPageProps): ReactElement => {
...
                         <Link
                             role="button"
                             href={
-                                returnToTemplateOverview
-                                    ? (queryParams["return"] as string)
-                                    : `${queryParams["return"] as string}/${pageState.disruptionId}${
-                                          queryParams["template"] ? "?template=true" : ""
-                                      }`
+                                returnToTemplateOverview || isTemplate === "true"
+                                    ? `${queryParams["return"] as string}/${pageState.disruptionId || ""}?template=true`
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/create-disruption/[disruptionId].page.tsx:652]: done

File: site/pages/create-disruption/[disruptionId].page.tsx [hunk 12]
```diff
@@ -637,27 +644,36 @@ const CreateDisruption = (props: DisruptionPageProps): ReactElement => {
...
+                                    : `${queryParams["return"] as string}/${pageState.disruptionId || ""}`
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/create-disruption/[disruptionId].page.tsx:653]: done

File: site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx [hunk 13]
```diff
@@ -32,6 +32,7 @@ const TypeOfConsequence = (props: ConsequenceTypePageProps): ReactElement => {
     const returnToTemplateOverview = returnTemplateOverview(queryParams);
 
     const isTemplate = (queryParams["template"] as string) || "";
+    const returnPath = (queryParams["return"] as string) || "";
```
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx:35]: done

File: site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx (hunk 1 above)
Comment (DeannaSharma) [reply site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx:72]: done

File: site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (DeannaSharma) [reply site/pages/type-of-consequence/[disruptionId]/[consequenceIndex].page.tsx:22]: IDE was reading from wrong config file - reverted change

**Review by ljones92 — APPROVED** [review]

**Pull request merged**